server/routes/receiveData/index.js

```diff
@@ -1,10 +1,487 @@
+	.then(function(obj){
+		tid = obj.tid
+		return savePreamble(rawJSON.preamble, tid)
+	})
+	.then(function() {
+		return saveErrorInfo(rawJSON.error_info, tid)
+	}) 
+	.then(function() {
+		return saveCurrentData(rawJSON.current_data_info, tid)
+	})
+	.then(function() {
+		return saveAttitudeData(rawJSON.data_attitude, tid)
+	})
+	.then(function() {
+		return saveDataFlashBurst(rawJSON.data_flash_burst, tid)
+	})
+	.then(function() {
+		return saveDataRadioTransmission(rawJSON.data_radio_transmission, tid)
+	})
+	.then(function(){
+		return res.sendStatus(200)
+	})
+	.catch(function(error){
+		console.error(error)
+		return res.sendStatus(500)
+	});
+}
 
-});
+function saveRaw(rawJSON) {
+	return (Raw.build({
+		raw: JSON.stringify(rawJSON)
+	}).save())
+}
+
+function savePreamble(preamble, tid) {
+	return (Preamble.build({
+		callsign: preamble.callsign,
+		timestamp: preamble.timestamp,
+		messageAndOpStates: preamble.message_and_op_states,
+		bytesInError: preamble.bytes_in_error,
+		bytesInMessage: preamble.bytes_in_message,
+		tid: tid
+	}).save())
+}
+
+function saveErrorInfo(errorInfo, tid) {
+	var promises = []
+	var errorArray = errorInfo.error
+	for(var i = 0; i < errorArray.length; i++){
+		promises.push(ErrorInfo.build({
+			index: i,
+			errorCode: errorArray[i],
+			tid: tid
+		}).save())
+	}
+	return Promise.all(promises)
+}
+
+function saveCurrentData(currentData, tid) {
+	var promises = []
+	
+	// Battery Data
+	saveCurrBatteryChargingAnalogVoltage(currentData.battery_charging_analog_voltages, tid, promises)
+	saveCurrBatteryChargingDigitalSignal(currentData.battery_charging_digital_signals, tid, promises)
+    saveCurrBatteryTemperature(currentData.battery_temperature, tid, promises)
+
+    // IMU Data
+    saveCurrImuAccelerometer(currentData.imu_accelerometer, tid, promises)
+    saveCurrImuGyroscope(currentData.imu_gyroscope, tid, promises)
+    saveCurrImuMagnetometer(currentData.imu_magnetometer, tid, promises)
+
+    // Lastflash Data
+    saveCurrLastFlashLedCurrent(currentData.last_flash_led_current, tid, promises)
+    saveCurrLastFlashLedTemperature(currentData.last_flash_led_temperature, tid, promises)
+    saveCurrLastFlashLifepoCurrent(currentData.last_flash_lifepo_currents, tid, promises)
+
+    // LiON Data
+    saveCurrLionCurrent(currentData.lion_currents, tid, promises)
+    saveCurrLionDischarge(currentData.lion_discharge, tid, promises)
+    saveCurrLionVoltage(currentData.lion_voltages, tid, promises)
+
+    // Radio Data
+	saveCurrRadioTemperature(currentData.radio_temperature, tid, promises)
+	saveCurrRadioVoltage(currentData.radio_voltage, tid, promises)
+
+	// Other
+	saveCurrDigitalOutput(currentData.digital_output, tid, promises)
+	saveCurrIrReading(currentData.ir_readings, tid, promises)
+	saveCurrPhotoDiode(currentData.photo_diode, tid, promises)
+
+	return Promise.all(promises)
+}
+
+function saveCurrBatteryChargingAnalogVoltage(voltages, tid, promises) {
+	for(var i = 0; i < voltages.length; i++){
```

Comment:
dont pass the promises array into these async functions. define it in here and return its `Promise.all()` you actually did it correctly in `saveErrorInfo`